api/src/main/java/org/openmrs/module/reporting/report/renderer/ReportTemplateRenderer.java

```diff
@@ -222,17 +222,22 @@ public String getExpressionSuffix(ReportDesign design) {
 	 * @return the value for the report template replacement given the initial value
 	 */
 	public Object getReplacementValue(Object initialValue) {
+
 		Object replacementValue = "";
-		if (initialValue != null) { 
-			if (initialValue instanceof Cohort) {
-				replacementValue = new Integer(((Cohort) initialValue).size());
-			} 
-			else if (initialValue instanceof IndicatorResult) {
-				replacementValue = new Double(((IndicatorResult) initialValue).getValue().doubleValue());
-			}
-			else {
-				replacementValue = initialValue;
+		if (initialValue != null) {
+			 if (initialValue instanceof IndicatorResult) {
+				IndicatorResult ir = (IndicatorResult) initialValue;
+				if(ir.getValue() instanceof Integer)
+				{
+					replacementValue = (Integer) ir.getValue();
+				}
+				else if(ir.getValue() instanceof Double) {
+					replacementValue = (Double) ir.getValue();
+				}

```

Comment:
What about an else case here?  What if it is an IndicatorResult but with a value other than Integer or Double?  You need to handle that case.